zaza/openstack/charm_tests/rabbitmq_server/tests.py

```diff
@@ -57,6 +58,41 @@ def _retry_get_amqp_message(self, check_unit, ssl=None, port=None):
                                                   ssl=ssl,
                                                   port=port)
 
+    @tenacity.retry(reraise=True, stop=tenacity.stop_after_attempt(30),
+                    retry=tenacity.retry_if_exception_type(AssertionError))
```

Comment:
It seems a little odd to me to use tenacity to implement a search.  Wouldn't a while loop just be easier until it runs out of messages on the queue?  The delay is implemented in `_retry_get_amqp_message`, so maybe tenacity is just complicating things here?